drivers/flash/nrf_qspi_nor.c

```diff
@@ -45,6 +45,19 @@ BUILD_ASSERT(INST_0_SCK_FREQUENCY >= (NRF_QSPI_BASE_CLOCK_FREQ / 16),
 
 #define WORD_SIZE 4
 
+#if DT_NODE_EXISTS(sram0)
+#define NRF_SOC_SRAM_BEGIN DT_REG_ADDR(DT_NODELABEL(sram0))
+#elif DT_NODE_EXISTS(sram1)
+#if DT_NODE_EXISTS(sram0_shared)
```

Comment:
sram0_shared is, effectively, an SRAM partition, not a physical RAM node.
I understand why you use this, though: because nrF5340 NETWORK uses the sram0_shared.
But this is ugly, IMO.

In my opinion the DTS node sram0 should be still defined for the NETWORK core (but not used, of course). It represents physical SoC memory and I don't see why the NETWORK core should not know about it.

So at the end of the day, both cores could have the sram0 symbol defined, and then this code here would only need to check for sram0 and sram1.

Reply:
I added the sram0 node to NETWORK core DTS. Is that what you want?